src/test/java/com/uber/cadence/workflow/MetricsTest.java

```diff
@@ -0,0 +1,87 @@
+package com.uber.cadence.workflow;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+import com.uber.cadence.client.WorkflowClient;
+import com.uber.cadence.client.WorkflowOptions;
+import com.uber.cadence.testing.TestEnvironmentOptions;
+import com.uber.cadence.testing.TestEnvironmentOptions.Builder;
+import com.uber.cadence.testing.TestWorkflowEnvironment;
+import com.uber.cadence.worker.Worker;
+import com.uber.m3.tally.RootScopeBuilder;
+import com.uber.m3.tally.Scope;
+import com.uber.m3.tally.StatsReporter;
+import java.time.Duration;
+import org.junit.Test;
+
+public class MetricsTest {
+  private static final String taskList = "metrics-test";
+
+  public interface TestWorkflow {
+    @WorkflowMethod
+    void execute();
+  }
+
+  public static class TestMetricsInWorkflow implements TestWorkflow {
+    @Override
+    public void execute() {
+      Workflow.getMetricsScope().counter("test-started").inc(1);
+
+      ChildWorkflowOptions options =
+          new ChildWorkflowOptions.Builder().setTaskList(taskList).build();
+      TestChildWorkflow workflow = Workflow.newChildWorkflowStub(TestChildWorkflow.class, options);
+      workflow.executeChild();
+
+      Workflow.getMetricsScope().counter("test-done").inc(1);
```

Comment:
Add a test of the workflow sleeping for some time and a metric measuring the latency of the sleep.

Reply:
Done.